website/static/js/contribRemover.js

```diff
@@ -0,0 +1,280 @@
+                                break;
+                            }
+                        }
+                        var registeredContributors = node.registeredContributors;
+                        var adminContributors = node.adminContributors;
+                        var visibleContributors = node.visibleContributors;
+                        var contributorIndex = node.registeredContributors.indexOf(contributorId);
+                        var adminIndex = adminContributors.indexOf(contributorId);
+                        var visibleIndex = visibleContributors.indexOf(contributorId);
+
+                        if (contributorIndex > -1) {
+                            registeredContributors.splice(contributorIndex, 1);
+                        }
+                        if (adminIndex > -1) {
+                            adminContributors.splice(adminIndex, 1);
+                        }
+                        if (visibleIndex > -1) {
+                            visibleContributors.splice(visibleIndex, 1);
+                        }
+                        self.canRemoveAdmin = adminContributors.length > 0;
+                        self.canRemoveVisible = visibleContributors.length > 0;
+                        self.canRemoveRegistered = registeredContributors.length > 0 ;
+                        canRemoveNodes[key] = (registeredContributors.length > 0 && adminContributors.length > 0 && visibleContributors.length > 0 && contributorOnNode);
+                    }
+                    else {
+                        canRemoveNodes[key] = false;
+                    }
+                }
+            }
+            return canRemoveNodes;
+        });
+
+        self.removeSelf = ko.computed(function() {
+            var currentUser = window.contextVars.currentUser.id;
+            if (self.contributorToRemove().id === currentUser) {
+                return true;
+            }
+            else {
+                return false;
+            }
+        });
+
+        self.canRemoveNode = ko.computed(function() {
+            return self.canRemoveNodes()[self.nodeId];
+        });
+
+        self.canRemoveNodesLength = ko.computed(function() {
+            var canRemoveNodeLengthLocal = 0;
+            for (var key in self.canRemoveNodes()) {
+                if (self.canRemoveNodes()[key]) {
+                    canRemoveNodeLengthLocal++;
+                }
+            }
+            return canRemoveNodeLengthLocal;
+        });
+
+        self.hasChildrenToRemove = ko.computed(function() {
+            //if there is more then one node to remove, then show a simplified page
+            if (self.canRemoveNodesLength() > 1) {
+                self.page(self.REMOVE);
+                return true;
+            }
+            else {
+                self.page(self.REMOVE_NO_CHILDREN);
+                return false;
+            }
+        });
+
+        self.modalSize = ko.pureComputed(function() {
+            var self = this;
```

Comment:
Minor: you've already saved `self = this` in the outer scope, so this line should be unnecessary.